File: Selenium/Activities/Activity5.py
from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://www.training-support.net/selenium/input-events")
    print("Title of page: ", driver.title)
    action = ActionChains(driver)
    action.click().pause(2).perform()
    print("Side active on left click: ", driver.find_element(By.XPATH, "//div[@class='active']").text)
    action.double_click().pause(2).perform()
    print("Side active on double click: ", driver.find_element(By.XPATH, "//div[@class='active']").text)
    action.context_click().pause(2).perform()
    print("Side active on right click: ", driver.find_element(By.XPATH, "//div[@class='active']").text)
    driver.quit()

File: Selenium/Activities/Activity5.java
package Activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;


public class Activity5 {
    public static void main(String[] args) throws InterruptedException
    {
        WebDriverManager.firefoxdriver().setup();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        WebDriver newDriver = new FirefoxDriver();
        newDriver.get("https://www.training-support.net/selenium/input-events");
        System.out.println("Title: " + newDriver.getTitle());
        Actions action = new Actions(newDriver);
        action.click().pause(1000).build().perform();
        WebElement leftEle = newDriver.findElement(By.xpath("//div[contains(@class,'active')]"));
        System.out.println("Side value on left click is: "+ leftEle.getText());
        action.doubleClick().pause(1000).build().perform();
        System.out.println("Side value on double click is: "+ newDriver.findElement(By.className("active")).getText());
        action.contextClick().pause(1000).build().perform();
        System.out.println("Side value on right click is: "+ newDriver.findElement(By.className("active")).getText());
        newDriver.quit();
    }
}
